File: back_end/clean_fish_supermarket/src/main/java/com/example/clean_fish_supermarket/service/impl/ProductService.java
package com.example.clean_fish_supermarket.service.impl;

import com.example.clean_fish_supermarket.model.Product;
import com.example.clean_fish_supermarket.repository.IProductRepository;
import com.example.clean_fish_supermarket.service.IProductService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductService implements IProductService {
    @Autowired
    private IProductRepository productRepository;
    @Override
    public List<Product> getLimitProductByType(int idType, int limit) {
        return productRepository.findLimitByType(idType, limit);
    }

    @Override
    public Product getProductById(int idProduct) {
        return productRepository.findByIdProductAndIsDeleteProductIsFalse(idProduct);
    }

    @Override
    public int countProductByIdProduct(int idType) {
        return productRepository.countProductByIdProduct(idType);
    }

    @Override
    public List<Product> searchByName(String name, int limit) {
        return productRepository.searchByName(name,limit);
    }

    @Override
    public int countSearch(String name) {
        return productRepository.countSearch(name);
    }

    @Override
    public List<Product> getListProductBetweenByPrice(int begin, int end, int idType, int limit) {
        return productRepository.getListProductBetweenByPrice(begin,end,idType,limit);
    }

    @Override
    public int countProductBetweenByPrice(int begin, int end, int idType) {
        return productRepository.countProductBetweenByPrice(begin,end,idType) ;
    }


}

File: front_end/fresh_fish_supermarket/src/service/productService.js
import axios from "axios";

export const getLimitListProductByType = async(idType,limit)=>{
   const res = await axios.get(`http://localhost:8080/product/get/${idType}/${limit}`)
return res.data;
}

export const getProductById = async(id)=>{
   const res = await axios.get(`http://localhost:8080/product/${id}`)
return res.data;
}

export const countProductByIdType = async(idType)=>{
   const res = await axios.get(`http://localhost:8080/product/count/${idType}`)
return res.data;
}

export const searchByName = async(name,limit)=>{
   const res = await axios.get(`http://localhost:8080/product/search/${name}/${limit}`)
return res.data;
}

export const countSearch = async(name)=>{
   const res = await axios.get(`http://localhost:8080/product/count-search/${name}`)
return res.data;
}

export const countSearchBetween = async(begin,end,idType)=>{
   const res = await axios.get(`http://localhost:8080/product/count-between/${begin}/${end}/${idType}`)
return res.data;
}

export const getListProductBetweenByPrice= async(begin,end,idType,limit)=>{
   const res = await axios.get(`http://localhost:8080/product/between/${begin}/${end}/${idType}/${limit}`)
return res.data;
}